### PR TITLE
Terraform 0.14 upgrade

### PR DESCRIPTION
**Pull request by Nuru**
## what
- Upgrade to support Terraform 0.14 and bring up to current Cloud Posse standard
- Remove IAM role and policies

## why
- Support Terraform 0.14
- Not needed now that we have service-linked roles
- Closes #25

### CODE REVIEW

**Review by aknysh — APPROVED** [review]

**Review by aknysh — CHANGES_REQUESTED** [review]
need to fix 
```
var.enabled is null
```

File: README.yaml [hunk 1]
```diff
@@ -51,7 +51,9 @@ description: |-
 usage: |-
   ```hcl
   module "dynamodb_autoscaler" {
-    source                       = "git::https://github.com/cloudposse/terraform-aws-dynamodb-autoscaler.git?ref=master"
+    source = "cloudposse/dynamodb-autoscaler/aws"
```
Comment (osterman) [inline README.yaml:54]: Our regex should capture the whitespace to preserve the formatting

Comment (Nuru) [reply README.yaml:54]: With the comment below it, the example is formatted as `terraform fmt` would format it.

Comment (osterman) [reply README.yaml:54]: Technically correct ;)

**Review by osterman — APPROVED** [review]

**Pull request merged**